packages/cli/src/commands/validator/status.ts

```diff
@@ -4,108 +4,128 @@ import { concurrentMap } from '@celo/utils/lib/async'
-      exclusive: ['validator'],
+      exclusive: ['validator', 'all'],
     }),
-    validator: Flags.address({
-      description: 'address of the validator to check if elected and validating',
-      exclusive: ['signer'],
+    all: flags.boolean({
+      description: 'get the status of all registered validators',
+      exclusive: ['validator', 'signer'],
     }),
     lookback: flags.integer({
       description: 'how many blocks to look back for signer activity',
       default: 100,
     }),
+    'no-truncate': flags.boolean({
+      description: "Don't truncate fields to fit line",
+      required: false,
+    }),
   }
 
   static examples = [
     'status --validator 0x5409ED021D9299bf6814279A6A1411A7e866A631',
-    'status --signer 0x738337030fAeb1E805253228881d844b5332fB4c',
-    'status --signer 0x738337030fAeb1E805253228881d844b5332fB4c --lookback 100',
+    'status --all --lookback 100',
   ]
 
   requireSynced = true
 
+  async getStatus(
+    signer: Address,
+    blocks: Block[],
+    electedSigners: Address[],
+    frontRunnerSigners: Address[]
+  ): Promise<ValidatorStatusEntry> {
+    const accounts = await this.kit.contracts.getAccounts()
+    const validator = await accounts.signerToAccount(signer)
+    const name = (await accounts.getName(validator)) || ''
+    const electedIndex = electedSigners.map((a) => eqAddress(a, signer)).indexOf(true)
+    const frontRunnerIndex = frontRunnerSigners.map((a) => eqAddress(a, signer)).indexOf(true)
+    const proposedCount = blocks.filter((b) => b.miner === signer).length
+    let signedCount = 0
+    if (electedIndex >= 0) {
+      signedCount = blocks.filter((b) =>
+        bitIsSet(parseBlockExtraData(b.extraData).parentAggregatedSeal.bitmap, electedIndex)
+      ).length
+    }
+    return {
+      name,
+      address: validator,
+      signer,
+      elected: electedIndex >= 0,
+      frontRunner: frontRunnerIndex >= 0,
+      proposed: proposedCount,
+      signatures: signedCount / blocks.length,
+    }
+  }
+
   async run() {
     const res = this.parse(ValidatorStatus)
-
-    // Check that the specified validator or signer meets the validator requirements.
-    const checker = newCheckBuilder(this, res.flags.signer)
-    if (res.flags.validator) {
-      const account = res.flags.validator
-      checker
-        .isAccount(account)
-        .isValidator(account)
-        .meetsValidatorBalanceRequirements(account)
-    } else if (res.flags.signer) {
-      checker
-        .isSignerOrAccount()
-        .signerMeetsValidatorBalanceRequirements()
-        .signerAccountIsValidator()
+    const accounts = await this.kit.contracts.getAccounts()
+    const validators = await this.kit.contracts.getValidators()
+    let signers: string[] = []
+    const checker = newCheckBuilder(this)
+    if (res.flags.signer) {
+      signers = [res.flags.signer]
+      const validator = await accounts.signerToAccount(res.flags.signer)
```

Comment:
Making this call before checking if it is a valid account means that if the address is not a signer or validator the response it `Error: Returned values aren't valid, did it run Out of Gas?`. I think we should try to avoid that.